llvm/lib/Target/AArch64/AArch64AsmPrinter.cpp

```diff
@@ -1243,26 +1243,41 @@ void AArch64AsmPrinter::EmitInstruction(const MachineInstr *MI) {
 
       ErrorOr<std::unique_ptr<MemoryBuffer>> FileOrErr =
           MemoryBuffer::getFileOrSTDIN(Filename);
-      if (std::error_code EC = FileOrErr.getError()) {
-        auto Err = SMDiagnostic(Filename, SourceMgr::DK_Error,
-                                "Could not open input file: " + EC.message());
-      }
 
-      Expected<llvm::json::Value> E =
-          llvm::json::parse((FileOrErr.get())->getBuffer());
-      assert(E);
-      Optional<int64_t> Padding;
-      if (llvm::json::Object *O = E->getAsObject())
-        if (llvm::json::Object *Opts = O->getObject("aarch64"))
-          if (Padding = Opts->getInteger(MILabel->getName())) {
-            assert(Opts->get(MILabel->getName())->kind() ==
-                   llvm::json::Value::Number);
+      if (FileOrErr) {
```

Comment:
nitpicking here, but since you went into trouble fixing this, does it make sense to figure out and handle errors first in if statements and then proceed with the "happy path" after those, so that they are not nested one after the other?
I think that's the usual handling of `Expected` also shown in the [docs][1].
Just inverting the conditions should do it.

[1]: https://llvm.org/doxygen/classllvm_1_1Expected.html

Reply:
I have seen both ways of checking the conditions and it is not clear to me what patterns we should use :)

However, in this case you are definitely right, it will remove unnecessary nesting!